app/assets/javascripts/discourse/controllers/preferences/emails.js.es6

```diff
@@ -42,6 +43,76 @@ export default Ember.Controller.extend(PreferencesTabController, {
     { name: I18n.t("user.email_previous_replies.never"), value: 2 }
   ],
 
+  @computed()
+  emailMessageValue() {
+    return JSON.stringify(
```

Comment:
Converting the options to a JSON string and back seems a bit cumbersome here.

My suggestion would be to use a single string value could be any of the three values. In your `emailOptionsChanged` callback use a `switch` to set the appropriate values based on the string.

I am also curious if you considered using a single value for this setting on the server side rather than adding a second option, which seems simpler in many ways?

Reply:
I have been going back and forth between all three options (JSON conversion, setting a string value and updating the options server-side) and ended up with the JSON conversion because it seemed cleaner (i.e. less code). 

I just noticed that we have a "checkbox + dropdown" in the UI for Activity Summary (in the same screen) so I will try using that pattern for the email preferences, and do any conversion, if necessary, in `emailOptionsChanged`.